### ISSUE TITLE
Linter not triggered

### ISSUE THREAD

**Issue by gggritso** [AtomLinter/linter-htmlhint#33]
On Atom 1.0.11, with linter-htmlhint 0.1.1 even manually setting the syntax for a file to HTML doesn't trigger linting at all if the file's extension isn't "html".

**Comment by devurandom** [comment]
I even have an html file, which is not being linted by linter-htmllint inside Atom. When I invoke `htmlhint` manually on the commandline, I get an error message, though:
```
$ ~/.atom/packages/linter-htmlhint/node_modules/htmlhint/bin/htmlhint thefile.html 

  thefile.html
      L416 |             <h2>Backend</h3>
                                    ^ Tag must be paired, no start tag: [ </h3> ] (tag-pair)
      L419 |            </section>
                        ^ Tag must be paired, missing: [ </h2> ], start tag match failed [ <h2> ] on line 416. (tag-pair)

2 errors in 1 files
```

**Comment by johnwebbcole** [comment]
Do you have a .htmlhintrc file anywhere above your file?  If so, make sure it has "tag-pair": true.  Your example worked fine for me.

**Comment by jasonblanchard** [comment]
Similar issue here, linter-htmlhint 0.1.1 Atom 1.0.19, getting this behavior (no linting) even if the file extension is `.html`.

**Comment by jasonblanchard** [comment]
fwiw, looks like the `linter-htmlhint` package installed version `0.9.8` of `htmlhint` which was bumped 2 days ago https://github.com/yaniswang/HTMLHint/commit/895970d985434b0edd4433e0d0f1204cebedf597#diff-b9cfc7f2cdf78a7f4b91a753d10865a2, maybe the command output was changed?

**Comment by jasonblanchard** [comment]
Confirmed, downgraded to `htmlhint` `0.9.6` and `linter-htmlhint` is linting in Atom  now.

**Comment by devurandom** [comment]
Is a `.htmlhintrc` file a requirement to use linter-hmtlhint? I would assume that linter-htmlhint manages it's backend programmes itself, so I never have to see any of it...

**Comment by johnwebbcole** [comment]
I've merged @prosf pr with uses the new json output option on `htmlhint 0.9.12`.  See if `0.2.0` works.

**Comment by leosanvieira** [comment]
I also have this issue, linting not working on html files. Atom 1.6.2, linter-htmlhint 1.1.1.

**Comment by Arcanemagus** [comment]
@leosanvieira  can you poen Atom's Developer Tools and put a breakpoint [here](https://github.com/AtomLinter/linter-htmlhint/blob/v1.1.1/lib/index.js#L45) to see if the linter is actually being triggered but not giving any output?

**Comment by leosanvieira** [comment]
@Arcanemagus I did as you asked and nothing happened, it didn't even get to the breakpoint.

**Comment by Arcanemagus** [comment]
If you run the `Editor: Log Cursor Scopes` command from the Command Palette what do you get?

**Comment by leosanvieira** [comment]
![image](https://cloud.githubusercontent.com/assets/9674817/14367483/b154121a-fcee-11e5-9768-2856049f6c6e.png)

**Comment by Arcanemagus** [comment]
Hmmm, you don't have `linter` disabled, do you?

**Comment by leosanvieira** [comment]
![image](https://cloud.githubusercontent.com/assets/9674817/14368657/85e39c98-fcf5-11e5-9ed0-82771115e891.png)
No, as you can see above. i also have other linter dependent packages and all of them are working fine.
If it were just that, the solution would be too easy, don't you think?

**Comment by Arcanemagus** [comment]
You'd be surprised how often that is the case :wink:

What does your `config.cson` look like?

**Comment by leosanvieira** [comment]
"linter-htmlhint": {}

There's only this section.

**Comment by aeoril** [comment]
I am having the same issue.  Any new status on this?  I am on Windows 10:

```
  linter: {}
  "linter-htmlhint": {}
```

linter-eslint and linter-csslint work fine:

<img width="513" alt="2016-04-11" src="https://cloud.githubusercontent.com/assets/1267648/14449147/ac0fd3a8-0035-11e6-9f21-3500091bf781.png">

**Comment by johnwebbcole** [comment]
Try removing 
 "linter-htmlhint": {}

I think that is turning off all rules.

**Comment by aeoril** [comment]
@johnwebbcole tried that - restarted Atom too, and upgraded to 1.1.2 and restarted - nothing worked

**Comment by leosanvieira** [comment]
I installed Atom on another machine today and linter-htmlhint is working now somehow but it only activates when i save the .html file, it does not activate after opening the same file which contains html "errors" and also does not activate after making changes, only after saving as i said before. 
Bases on other linter-* plugins, i don't think that is how the default behavior should be like.
Debian Jessie
Atom 1.7.1
linter 1.11.4
linter-htmlhint 1.1.2
There are no sections in config.cson about linter or linter-htmlhint.

**Comment by Arcanemagus** [comment]
@leosanvieira There are two general "classes" of linter providers, those that can run on the current file contents, and those that only operate on a file saved to disk. Currently at least this linter belongs to that second class and will only run when the file is saved.

There is [an issue](https://github.com/steelbrain/linter/issues/1052) open on `linter` for enabling triggering a lint on file open, you may want to subscribe to that.

**Comment by Arcanemagus** [comment]
@aeoril Were you by chance on the v1.7.0 betas when you posted that? There was a bug with Node process execution on Windows that v1.7.1 (and v1.8.0-beta1) fixed.

@gggritso Are you still having issues?

**Comment by gggritso** [comment]
I am not! After updating everything, checking the settings and correcting my `.rc` everything is working as expected. Thank you for your help!

**Comment by Arcanemagus** [comment]
Marking as closed due to a lack of response from @aeoril.

If you are still having issues with this please file a new issue!
